### PR TITLE
feat(data-exploration): improve data table crashes

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

The data exploration table would crash in weird ways when adding columns that didn't make sense. You'd have to go and edit the query to recover from this state:

## Changes

Now it either shows a full page "Error", or then uses a **antd Tag** to show the error in the table.

The error component is stolen from insights. I also added a "no results" message, again from insights:

image

## Out of scope

There's a separate PR out that exposes the actual errors themselves to the users:

https://github.com/PostHog/posthog/pull/13703

This PR wants to do the minimum work in order to let the users recover from a common scenario of "I added faulty hogql, how do I recover?".

## How did you test this code?

Made it crash.

### CODE REVIEW

File: frontend/src/queries/nodes/DataTable/renderColumn.tsx [hunk 1]
```diff
@@ -21,7 +24,11 @@ export function renderColumn(
     setQuery?: (query: DataTableNode) => void,
     context?: QueryContext
 ): JSX.Element | string {
-    if (key === 'event' && isEventsQuery(query.source)) {
+    if (value === loadingColumn) {
+        return <Spinner />
+    } else if (value === errorColumn) {
+        return <Tag color="red">Error</Tag>
```
Comment (Twixes) [inline frontend/src/queries/nodes/DataTable/renderColumn.tsx:30]: You're giving me extra work 😄 Why not `LemonTag`?

File: frontend/src/queries/nodes/DataTable/dataTableLogic.ts [hunk 2]
```diff
@@ -84,16 +114,26 @@ export const dataTableLogic = kea<dataTableLogicType>([
...
                 }
                 return response && 'results' in response ? (response as any).results ?? null : null
             },
         ],
+        isRowHighlighted: [
+            (s) => [s.highlightedRows],
+            (highlightedRows) =>
+                (row: any[]): boolean =>
+                    row
+                        ? highlightedRows.has('__originalResultRow' in row ? (row as any).__originalResultRow : row)
```
Comment (Twixes) [inline frontend/src/queries/nodes/DataTable/dataTableLogic.ts:134]: Can't we add `__originalResultRow` to the type?

File: frontend/src/queries/nodes/DataTable/dataTableLogic.ts [hunk 3]
```diff
@@ -84,16 +114,26 @@ export const dataTableLogic = kea<dataTableLogicType>([
...
                             return newResults
+                        } else {
+                            return results.map((result) => convertResultToDisplayedColumns(result))
                         }
                     }
                 }
                 return response && 'results' in response ? (response as any).results ?? null : null
             },
         ],
+        isRowHighlighted: [
```
Comment (Twixes) [inline frontend/src/queries/nodes/DataTable/dataTableLogic.ts:129]: Hm I can't quite grasp the purpose of this highlighting from the code. Can you add a one-line comment explaining what rows we want highlighted?

File: frontend/src/queries/nodes/DataTable/renderColumn.tsx (hunk 1 above)
Comment (mariusandra) [reply frontend/src/queries/nodes/DataTable/renderColumn.tsx]: I didn't find it 🤪 

image

File: frontend/src/queries/nodes/DataTable/dataTableLogic.ts (hunk 3 above)
Comment (mariusandra) [reply frontend/src/queries/nodes/DataTable/dataTableLogic.ts]: The row data structure refactor removed the need for this proxying. `yeet!`

**Review by Twixes — COMMENTED** [review]
Noticed a slightly weird thing UX-wise:
If I first add an invalid expression column (e.g. `!!!`) and then a valid one (e.g. `properties.$geoip_country_code`), both will show up as "Error"ing. And even when I delete the invalid one, the valid one will still be all "Error"s. If we could somehow determine which column is the offending one, this would be more intuitive.
Can we do this easily, considering we already catch problems like the column not existing in `property_access_to_clickhouse`, or would that be overkill for now?

**Comment by mariusandra** [comment]
I went for a slightly simpler solution for now: just remove the results when we have an error. This also makes it easier to actually [show errors](https://github.com/PostHog/posthog/pull/13703) when they occur.

**Review by Twixes — APPROVED** [review]
Let's do this

**Pull request merged**